src/lib.rs

```diff
@@ -360,6 +360,23 @@ impl Serialize for Target {
     }
 }
 
+fn warn_on_failure(target: &Target, toolchain: &str, msg_info: MessageInfo) -> Result<()> {
+    let rust_std = format!("rust-std-{target}");
+    if target.is_builtin() {
+        let component = rustup::check_component(&rust_std, toolchain, msg_info)?;
+        if component.is_not_available() {
+            shell::warn(format!("rust-std is not available for {target}"), msg_info)?;
+            shell::note(
+                "you may need to build components for the std library `-Z build-std=<components>`
+              the available components are core, std, alloc, and proc_macro
+              if no components are provided, cargo will build all components",
+                msg_info,
+            )?;
```

Comment:
```suggestion
            shell::note(
                format_args("you may need to build components for the target via `-Z build-std=<components>` or in your cross configuration specify `target.{target}.build-std`
              the available components are core, std, alloc, and proc_macro"),
                msg_info,
            )?;
```

Reply:
don't think the last part is needed, first reading it I was confused as to what it was referring to.